state/store.go

```diff
@@ -94,6 +98,9 @@ func saveState(db dbm.DB, state State, key []byte) {
 		lastHeightVoteChanged := int64(1)
 		saveValidatorsInfo(db, nextHeight, lastHeightVoteChanged, state.Validators)
 	}
+	//save software info
+	saveSoftware(db, nextHeight, state.Version.Software)
```

Comment:
Why does this need to be saved independently of the state? The point of putting the version at the beginning of the state is so we could always read it out using a simple structure even if we change the rest of the fields in the state.